lib/ocpp/common/types.cpp

```diff
@@ -53,8 +43,7 @@ void DateTimeImpl::from_rfc3339(const std::string& timepoint_str) {
             in.seekg(0);
             in >> date::parse("%FT%T", this->timepoint);
             if (in.fail()) {
-                EVLOG_error << "Timepoint string parsing failed. Could not convert: \"" << timepoint_str
-                            << "\" into DateTime.";
+                throw TimePointParseException(timepoint_str);
```

Comment:
Hmm now you added a throw where the software did not crash in the past. Did you check everywhere this is used and provide a try / catch around it if needed?

Reply:
Well, the `std::string` constructor calls only when json is converted to the `DateTime` type or in database conversions. In other places the constructor is not (should not be) called. So then it should be caught either by `TimePointParseException` or `std::exception`, that is done. From my side and for 2.0.1, I don't see the missing places. @hikinggrass can approve as well for 1.6